engine/access/subscription/subscription.go

```diff
@@ -19,7 +41,29 @@ import (
 // All other errors are considered exceptions
 type GetDataByHeightFunc func(ctx context.Context, height uint64) (interface{}, error)
 
-var _ state_stream.Subscription = (*SubscriptionImpl)(nil)
+// Subscription represents a streaming request, and handles the communication between the grpc handler
+// and the backend implementation.
+type Subscription interface {
+	// ID returns the unique identifier for this subscription used for logging
+	ID() string
+
+	// Channel returns the channel from which subscription data can be read
+	Channel() <-chan interface{}
+
+	// Err returns the error that caused the subscription to fail
+	Err() error
+}
+
+// Streamable represents a subscription that can be streamed.
+type Streamable interface {
```

Comment:
I understand we are moving around things from different package but can you extend documentation for functions in this interface so it's clear how to use it. Thank you.